rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -335,75 +335,77 @@ impl Expr {
         }
     }
 
-    /// Equal
-    pub fn eq(&self, other: Expr) -> Expr {
-        binary_expr(self.clone(), Operator::Eq, other)
+    /// Return `self == other`
+    pub fn eq(self, other: Expr) -> Expr {
```

Comment:
The change is to have all these functions take `self` rather than `&self`. 

Given they almost always are used like `let expr = lit("foo").eq(lit("bar"))` self is not reused.

Furthermore, the `clone()` is actually doing a deep clone, so in the above example `lit("foo")` gets copied twice.